src/mib_props_supp.c

```diff
@@ -0,0 +1,98 @@
+#include "mib_props_supp.h"
+#include "mib_header_DAC.h"
+#include "mib_header_MQ1.h"
+#include "mib_macros.h"
+#include "mib_utils.h"
+#include "mq1_quad.h"
+#include "mq1_single.h"
+#include "read_mq1_headers.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Mainly this is just for easy access to mq1_fields and dac
+ */
+
+info *mq1_fields_iter(MQ1_fields *fields_struct, size_t *out_count)
+{
+  if (!fields_struct || !out_count)
+    return NULL;
+
+  const size_t NUM_FIELDS = 19;
+
+  info *fields = malloc(NUM_FIELDS * sizeof(info));
+  if (!fields)
+    return NULL;
+
+  size_t i = 0;
+
+  fields[i++] = (info) {"max_length", &fields_struct->max_length};
+  fields[i++] = (info) {"sequence_number", fields_struct->sequence_number};
+  fields[i++] = (info) {"header_bytes", fields_struct->header_bytes};
+  fields[i++] = (info) {"num_chips", fields_struct->num_chips};
+  fields[i++] = (info) {"det_x", fields_struct->det_x};
+  fields[i++] = (info) {"det_y", fields_struct->det_y};
+  fields[i++] = (info) {"pixel_depth", fields_struct->pixel_depth};
+  fields[i++] = (info) {"sensor_layout", fields_struct->sensor_layout};
+  fields[i++] = (info) {"chip_select", fields_struct->chip_select};
+  fields[i++] = (info) {"timestamp", fields_struct->timestamp};
+  fields[i++] = (info) {"exposure_time_s", fields_struct->exposure_time_s};
+  fields[i++] = (info) {"counter", fields_struct->counter};
+  fields[i++] = (info) {"colour_mode", fields_struct->colour_mode};
+  fields[i++] = (info) {"gain_mode", fields_struct->gain_mode};
+  fields[i++] = (info) {"threshold", fields_struct->threshold};
+  fields[i++] =
+    (info) {"header_extension_id", fields_struct->header_extension_id};
+  fields[i++] =
+    (info) {"extended_timestamp", fields_struct->extended_timestamp};
+  fields[i++] = (info) {"exposure_time_ns", fields_struct->exposure_time_ns};
+  fields[i++] = (info) {"bit_depth", fields_struct->bit_depth};
```

Comment:
Can we just use `field[0]`, `field[1]` etc if the number of field is a constant? If the variable `i` serve as counting, a check on the array bound is needed.